custom-pages/gadgets/reports/reports.js

```diff
@@ -1,13 +1,192 @@
+        make_xlsx_lib(XLSX) //eslint-disable-line no-undef
+    });
     const html = require('text!./templates/reports.html');
     
     const Empty = require('ratchet/dynamic/empty');
     
     const UI = require('ui');
-    
+    const $ = require('jquery')
+
+    const [PAGE, PRODUCT, SKU, PRICESHEET] = [
+        'page',
+        'product',
+        'sku',
+        'price-sheet'
+    ]
+    const ReportTypes = [PAGE, PRODUCT, SKU, PRICESHEET]
+
+    const QueryTypes = {
+        [PAGE]: {
+            "$regex": "cricket:page(-.+)?"
+        },
+        [PRODUCT]: "cricket:product",
+        [SKU]: "cricket:sku",
+        [PRICESHEET]: "cricket:price-list"
+    }
+
     let branch;
+    let platform;
+
+    function buildQuery(reportType) {
+        let query = { }
+        if (reportType in QueryTypes) {
+            query._type = QueryTypes[reportType]
+        } else {
+            console.error('Invalid report type provided')
+        }
+
+        if (query._type) {
+            return query
+        } else {
+            return false
+        }
+    }
+
+    function genericErrorLoggerHalter(err) {
+        console.error(err)
+        return false
+    }
+
+    function buildWorksheet({nodes, reportType}) {
+        const topLevelFieldsToCopy = ["title", "active", "sol", "soli", "eol", "eoli"]
+        let filteredNodes = []
+
+        nodes = nodes.map((record) => {
+            let rec = {}
+
+            //copy common fields
+            topLevelFieldsToCopy.forEach(function (fieldName) {
+                rec[fieldName] = record[fieldName]
+            })
+
+            rec._type = record.getTypeQName()
+            let meta = record.getSystemMetadata()
+            rec["_system.created_on.timestamp"] = meta.created_on.timestamp
+            rec["_system.created_by"] = meta.created_by
+            rec["_system.edited_on.timestamp"] = meta.edited_on.timestamp
+            rec["_system.edited_by"] = meta.edited_by
+
+            //copy all skus' .skuId to top-level prop on rec
+            if (PRODUCT === reportType) {
+                record.skus.forEach(function (sku, index) {
+                    rec[`skus[${index}].skuId`] = sku.skuId
+                })
+            }
+
+            //copy over product page.urlList[0].url
+            if (PAGE === reportType) {
+                if (record.urlList && record.urlList[0]) {
+                    rec.url = record.urlList[0].url
+                }
+            }
+
+            //copy over sku.color[0].hexValue and displayName
+            if (SKU === reportType) {
+                rec.skuId = record.skuId
+
+                if (record.color && record.color[0]) {
+                    rec.hexValue = record.color[0].hexValue
+                    rec.displayName = record.color[0].displayName
+                }
+            }
+
+            //copy over priceSkuList[0].price[ 0-2 ].priceType and priceValue
+            //copy over priceSkuList[0].sku[0].sol, soli, eol, and eoli
```

Comment:
I do not see these properties copied

Reply:
Yep. Originally had these in the CSV but have removed from xlsx for now